nautobot/ipam/querysets.py

```diff
@@ -14,15 +61,74 @@ def annotate_tree(self):
-                'WHERE (U0."prefix" >> "ipam_prefix"."prefix" '
-                f'AND COALESCE(U0."vrf_id", \'{FAKE_UUID}\') = COALESCE("ipam_prefix"."vrf_id", \'{FAKE_UUID}\'))',
-                "children": 'SELECT COUNT(U1."prefix") AS "c" '
-                'FROM "ipam_prefix" U1 '
-                'WHERE (U1."prefix" << "ipam_prefix"."prefix" '
-                f'AND COALESCE(U1."vrf_id", \'{FAKE_UUID}\') = COALESCE("ipam_prefix"."vrf_id", \'{FAKE_UUID}\'))',
-            }
+
+        qs = self.annotate(
+            maybe_vrf=ExpressionWrapper(
+                Coalesce(F("vrf_id"), FAKE_UUID),
+                output_field=UUIDField(),
+            )
         )
+
+        return qs.annotate(
+            parents=RawSQL(
+                "SELECT COUNT(*) FROM ipam_prefix AS U0 "
+                "WHERE U0.prefix_length < ipam_prefix.prefix_length "
+                "AND U0.network <= ipam_prefix.network "
+                "AND U0.broadcast >= ipam_prefix.broadcast "
+                f"AND COALESCE(U0.vrf_id, '{FAKE_UUID}') >= COALESCE(ipam_prefix.vrf_id, '{FAKE_UUID}')",
+                (),
+            ),
+            children=RawSQL(
+                "SELECT COUNT(*) FROM ipam_prefix AS U1 "
+                "WHERE U1.prefix_length = ipam_prefix.prefix_length "
+                "AND U1.network >= ipam_prefix.network "
+                "AND U1.broadcast <= ipam_prefix.broadcast "
+                f"AND COALESCE(U1.vrf_id, '{FAKE_UUID}') <= COALESCE(ipam_prefix.vrf_id, '{FAKE_UUID}')",
+                (),
+            ),
+        )
+
+
+class IPAddressQuerySet(RestrictedQuerySet):
+    ip_family_map = {
+        4: IPV4_BYTE_LENGTH,
+        6: IPV6_BYTE_LENGTH,
+    }
+
+    def get_queryset(self):
+        """
+        By default, PostgreSQL will order INETs with shorter (larger) prefix lengths ahead of those with longer
+        (smaller) masks. This makes no sense when ordering IPs, which should be ordered solely by family and host
+        address. We can use HOST() to extract just the host portion of the address (ignoring its mask), but we must
+        then re-cast this value to INET() so that records will be ordered properly. We are essentially re-casting each
+        IP address as a /32 or /128.
+        """
+        return super().order_by("host")
+
+    def ip_family(self, family):
+        try:
+            byte_len = self.ip_family_map[family]
+        except KeyError:
+            raise ValueError("invalid IP family {}".format(family))
+
+        return self.annotate(address_len=Length(F("host"))).filter(address_len=byte_len)
+
+    def net_host_contained(self, network):
+        return self.filter(
+            host__lte=bytes(network.broadcast),
+            host__gte=bytes(network.network),
+        )
+
+    def net_contained_or_equal(self, network):
+        return self.filter(
+            host__lte=bytes(network.broadcast),
+            host__gte=bytes(network.host),
+        )
```

Comment:
Comments and unit tests requested here as well. :-) But this seems wrong - `net_host_contained` and `net_contained_or_equal` are meaning exactly the same thing?